File: examples/python/integration_check.py
"""Deterministic host check report and read-only recovery inspection; no live services."""
import asyncio
import json
from importlib.metadata import version
from purra.api import IntegrationCheck, check_integration, inspect_recovery
from purra.adapters import InMemoryAgentAdapters
from purra.contracts import RunCreateParams
from purra.events import AgentEvent
from purra.testing import assert_artifact_store_conforms

async def main():
    fixture = InMemoryAgentAdapters()
    async def storage_probe():
        await assert_artifact_store_conforms(artifacts=fixture.artifacts,
            claims=fixture.artifact_claims, maintenance=fixture.artifact_maintenance)
    report = await check_integration(component='purra', version=version('purra'),
        checks=[IntegrationCheck('storage', 'deterministic', storage_probe)])
    assert next(row for row in report['checks'] if row['capability']=='storage' and row['category']=='deterministic')['status'] == 'passed'
    run = await fixture.runs.begin(RunCreateParams(None, 'fixture', None), AgentEvent('run.started'))
    recovery = await inspect_recovery(fixture.runs, run.run_id)
    assert recovery['observations']['lease'] == 'unknown'
    print(json.dumps({'integration':report, 'recovery':recovery}, indent=2))

if __name__ == '__main__': asyncio.run(main())

File: typescript/examples/integration-check.ts
// Isolated deterministic store; assertions never operate on production data.
import { InMemoryRunRepository, assertRunRepositoryConforms, checkIntegration, inspectRecovery } from "purra";
const fixture = new InMemoryRunRepository();
const report = await checkIntegration({component:"purra",version:"0.5.0",
  checks:[{capability:"storage",category:"deterministic",probe:()=>assertRunRepositoryConforms(fixture)}]});
if (report.checks.find(row=>row.capability==="storage"&&row.category==="deterministic")?.status !== "passed") throw Error("fixture failed");
const recovery = await inspectRecovery(fixture,"conformance-run-1");
if (recovery.observations.lease !== "unknown") throw Error("missing evidence was inferred");
console.log(JSON.stringify({integration:report,recovery},null,2));
